docs/operators/csvParserOperator.md

```diff
@@ -0,0 +1,99 @@
+=============
+
+## Operator Objective
+This operator is designed to parse delimited records and construct a map or concrete java class also known as ["POJO"](https://en.wikipedia.org/wiki/Plain_Old_Java_Object) out of it. User need to provide the schema to describe the delimited data. Based on schema definition the operator will parse the incoming record to object map and POJO.  User can also provide constraints if any, in the schema. The supported constraints are listed in [constraints table](#constraints). The incoming record will be validated against those constraints. Valid records will be emitted as POJO / map while invalid ones are emitted on error port with error message.
+
+**Note**: field names of POJO must match field names in schema and in the same order as it appears in the incoming data.
+
+## Overview
+The operator is **stateless**, **idempotent**, **fault-tolerant** and **partitionable**.
+
+
+## <a name="props"></a>Properties of Csv Parser
+User need to set the schema which describes delimited data as well as specifies constraints on values if any.
+e.g.
+
+```xml
+{
+  "separator":",",
+  "quoteChar":"\"",
+  "fields":[
+    {
+      "name":"adId",
+      "type":"Integer",
+      "constraints":{
+         "required":"true"
+      }
+    },
+    {
+      "name":"adName",
+      "type":"String",
+      "constraints":{
+         "required":"true",
+         "pattern":"[a-z].*[a-z]$",
+         "maxLength":"10"
+      }
+    },
+    {
+      "name":"bidPrice",
+      "type":"Double",
+      "constraints":{
+         "required":"true",
+         "minValue":"0.1",
+         "maxValue":"3.2"
+      }
+    },
+    {
+      "name":"startDate",
+      "type":"Date",
+      "constraints":{
+         "format":"yyyy-MM-dd HH:mm:ss",
+         "locale":"IN"
+      }
+    }
+  ]
+}
+```
+
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *schema* | [Schema](https://github.com/apache/apex-malhar/blob/master/contrib/src/main/java/com/datatorrent/contrib/parser/DelimitedSchema.java)  describing delimited data | String | YES | N/A |
+ 
+## Platform Attributes that influences operator behavior
+
+| **Attribute** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *out.TUPLE_CLASS* | TUPLE_CLASS attribute on output port which tells operator the class of POJO which need to be emitted | Class| Yes |
+ 
+## <a name="constraints"></a>Schema Constraints 
+|**Type**| **Constraints** | **Description** |
+| -------- | -------- | -------- |
+| *required* | True or False | If mentioned, indicates that the data type constraints that follow are required. |
+| *equals* | True or False | If mentioned, indicates that the data string or value declared in the data type constraints must be an exact match with the specified value. `Note: This constraints is not applicable for data type boolean and date` |
```

Comment:
1. Type can be renamed to DataType. 
2. Required and equals should fall under constraints. These two are applicable for all datatypes so we need to mention that as well.
3. Required means the field is required. It cannot be blank/null. It may or may not satisfy other constraints ( like equals/minVal/maxVal etc )